pkg/driver/controller_test.go

```diff
@@ -432,166 +864,299 @@ func TestPickAvailabilityZone(t *testing.T) {
+				resp, err := awsDriver.CreateSnapshot(context.Background(), req)
+				if err != nil {
+					srvErr, ok := status.FromError(err)
+					if !ok {
+						t.Fatalf("Could not get error status code from error: %v", srvErr)
+					}
+				}
+				if expErrCode != codes.OK {
+					t.Fatalf("Expected error %v, got no error", expErrCode)
+				}
+				snap := resp.GetSnapshot()
+				if snap == nil {
+					t.Fatalf("Expected snapshot %v, got nil", expSnapshot)
+				}
+
+				mockCloud.EXPECT().GetSnapshotByName(gomock.Eq(ctx), gomock.Eq(extraReq.GetName())).Return(mockSnapshot, nil)
+				_, err = awsDriver.CreateSnapshot(ctx, extraReq)
+				if err != nil {
+					srvErr, ok := status.FromError(err)
+					if !ok {
+						t.Fatalf("Could not get error status code from error: %v", srvErr)
+					}
+					if srvErr.Code() != extraExpErrCode {
+						t.Fatalf("Expected error code %d, got %d message %s", expErrCode, srvErr.Code(), srvErr.Message())
+					}
+					return
+				}
+				if extraExpErrCode != codes.OK {
+					t.Fatalf("Expected error %v, got no error", extraExpErrCode)
+				}
 			},
-			expErrCode:      codes.OK,
-			extraExpErrCode: codes.AlreadyExists,
 		},
 		{
 			name: "success same name same volume ID",
-			req: &csi.CreateSnapshotRequest{
-				Name:           "test-snapshot",
-				Parameters:     nil,
-				SourceVolumeId: "vol-test",
-			},
-			extraReq: &csi.CreateSnapshotRequest{
-				Name:           "test-snapshot",
-				Parameters:     nil,
-				SourceVolumeId: "vol-test",
-			},
-			expSnapshot: &csi.Snapshot{
-				ReadyToUse: true,
+			testFunc: func(t *testing.T) {
+				req := &csi.CreateSnapshotRequest{
+					Name:           "test-snapshot",
+					Parameters:     nil,
+					SourceVolumeId: "vol-test",
+				}
+				extraReq := &csi.CreateSnapshotRequest{
+					Name:           "test-snapshot",
+					Parameters:     nil,
+					SourceVolumeId: "vol-test",
+				}
+				expSnapshot := &csi.Snapshot{
+					ReadyToUse: true,
+				}
+				expErrCode := codes.OK
+				extraExpErrCode := codes.OK
+
+				ctx := context.Background()
+				mockSnapshot := &cloud.Snapshot{
+					SnapshotID:     fmt.Sprintf("snapshot-%d", rand.New(rand.NewSource(time.Now().UnixNano())).Uint64()),
+					SourceVolumeID: req.SourceVolumeId,
+					Size:           1,
+					CreationTime:   time.Now(),
+				}
+				mockCtl := gomock.NewController(t)
+				defer mockCtl.Finish()
+
+				mockCloud := mocks.NewMockCloud(mockCtl)
+				mockCloud.EXPECT().GetSnapshotByName(gomock.Eq(ctx), gomock.Eq(req.GetName())).Return(nil, cloud.ErrNotFound)
+				mockCloud.EXPECT().CreateSnapshot(gomock.Eq(ctx), gomock.Eq(req.SourceVolumeId), gomock.Any()).Return(mockSnapshot, nil)
+
+				awsDriver := controllerService{cloud: mockCloud}
+				resp, err := awsDriver.CreateSnapshot(context.Background(), req)
+				if err != nil {
+					srvErr, ok := status.FromError(err)
+					if !ok {
+						t.Fatalf("Could not get error status code from error: %v", srvErr)
+					}
+				}
+				if expErrCode != codes.OK {
```

Comment:
This line is always true?
```suggestion
				if err != codes.OK {
```